language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ExtractToConfigurableCodeAction.java

```diff
@@ -0,0 +1,346 @@
+                    } else if (argKind == SyntaxKind.POSITIONAL_ARG) {
+                        argument = ((PositionalArgumentNode) argument).expression();
+                    } else {
+                        return Collections.emptyList();
+                    }
+
+                    if (!supportedExpressionList.contains(argument.kind())) {
+                        return Collections.emptyList();
+                    }
+                    Optional<TypeSymbol> argTypeSymbol = semanticModel.typeOf(argument);
+                    if (argTypeSymbol.isEmpty()) {
+                        return Collections.emptyList();
+                    }
+                    TypeSymbol argType = argTypeSymbol.get();
+                    if (!argType.subtypeOf(anydataType)) {
+                        return Collections.emptyList();
+                    }
+                    ParameterSymbol parameterSymbol = parameterSymbols.get(argIdx);
+                    String paramName = parameterSymbol.getName().orElse(CONFIGURABLE_NAME_PREFIX);
+                    String confName = getConfigurableNameUsingParamName(context, paramName);
+                    ConfigurableData configurableData = getConfigurableData(context);
+                    Position positionToAddConfVar = configurableData.position();
+                    boolean addNewLineAtStart = configurableData.addNewLineAtStart() && firstEdit;
+                    firstEdit = false;
+
+                   textEdits.addAll(getTextEdits(context, argument, argType, confName, positionToAddConfVar,
+                           addNewLineAtStart));
+                }
+            }
+            CodeAction codeAction = CodeActionUtil.createCodeAction(CommandConstants.EXTRACT_TO_CONFIGURABLE,
+                    textEdits, context.fileUri(), CodeActionKind.RefactorExtract);
+            return List.of(codeAction);
+        }
+
+        Optional<TypeSymbol> typeSymbol = semanticModel.typeOf(node);
+        if (typeSymbol.isEmpty() || typeSymbol.get().typeKind() == TypeDescKind.COMPILATION_ERROR) {
+            return Collections.emptyList();
+        }
+
+        String confName = getConfigurableName(context);
+        ConfigurableData configurableData = getConfigurableData(context);
+        Position positionToAddConfVar = configurableData.position();
+        boolean addNewLineAtStart = configurableData.addNewLineAtStart();
+
+        List<TextEdit> textEdits = getTextEdits(context, node, typeSymbol.get(), confName, positionToAddConfVar,
+                addNewLineAtStart);
+
+        CodeAction codeAction = CodeActionUtil.createCodeAction(CommandConstants.EXTRACT_TO_CONFIGURABLE,
+                textEdits, context.fileUri(), CodeActionKind.RefactorExtract);
+        CodeActionUtil.addRenamePopup(context, codeAction, CommandConstants.RENAME_COMMAND_TITLE_FOR_CONFIGURABLE,
+                getRenamePosition(textEdits.get(1).getRange(), addNewLineAtStart));
```

Comment:
Don't we have to make sure that `textEdits` has at least 1 change before calling `textEdits.get(1).getRange()`

Reply:
This is guaranteed that we always get only two items in the list.